oidc-authenticator/src/main/java/org/xwiki/contrib/oidc/auth/OIDCAuthServiceImpl.java

```diff
@@ -76,6 +106,32 @@ public class OIDCAuthServiceImpl extends XWikiAuthServiceImpl
 
     private ScriptContextManager scriptContextManager = Utils.getComponent(ScriptContextManager.class);
 
+    private Container container = Utils.getComponent(Container.class);
+
+    private OIDCUserStore store = Utils.getComponent(OIDCUserStore.class);;
+
+    private ConfigurableJWTProcessor<SecurityContext> jwtProcessor;
+
+    public OIDCAuthServiceImpl() {
+        jwtProcessor = new DefaultJWTProcessor<>();
+
+        try {
+            // Set the required "typ" header "at+jwt" for access tokens
+            jwtProcessor.setJWSTypeVerifier(new DefaultJOSEObjectTypeVerifier<>(new JOSEObjectType("JWT")));
+            JWKSource<SecurityContext> keySource;
+            keySource = JWKSourceBuilder.create(configuration.getClientProvider().getMetadata().getJWKSetURI().toURL())
+                    .retrying(true).build();
+
+            JWSAlgorithm expectedJWSAlg = JWSAlgorithm.RS256;
+            // Configure the JWT processor with a key selector to feed matching public
+            // RSA keys sourced from the JWK set URL
+            JWSKeySelector<SecurityContext> keySelector = new JWSVerificationKeySelector<>(expectedJWSAlg, keySource);
+            jwtProcessor.setJWSKeySelector(keySelector);
+        } catch (IOException | GeneralException | URISyntaxException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
```

Comment:
Better keep `OIDCAuthServiceImpl` complexity to a minimum and move things like this and `#checkAccessToken` in classes like `OIDCUserManager`, so the `users` field (or `OIDCManager`/`manager` when it's more generic than the authentication use case).